src/Mvc/Mvc.Core/src/Routing/ActionEndpointFactory.cs

```diff
@@ -56,6 +57,17 @@ public ActionEndpointFactory(RoutePatternTransformer routePatternTransformer)
                 throw new ArgumentNullException(nameof(conventions));
             }
 
+            if (createInertEndpoints)
+            {
+                // For each ActionDescriptor create a single 'inert' Endpoint without routing information.
+                //
+                // This makes those endpoints accessible for dynamic and fallback routing. We don't want to
+                // mix RouteEndpoint that is selected by routing, vs Endpoint that is selected by user code/policies
+                //
+                // This also helps us avoid ambiguities when an endpoint has multiple conventional routes.
+                endpoints.Add(CreateInertEndpoint(action, conventions));
+            }
+
```

Comment:
Aside from this all of the other changes in this file are refactoring to avoid duplication for the route endpoint and inert endpoint.